src/askui_runner/modules/core/infrastructure/files/askui.py

```diff
@@ -244,55 +211,65 @@ def _download_file(
                 if chunk:
                     f.write(chunk)
 
-        cloud_timestamp = (
-            datetime.strptime(last_modified, "%Y-%m-%dT%H:%M:%SZ")
-            .replace(tzinfo=timezone.utc)
-            .timestamp()
-        )
-        os.utime(path=local_file_path, times=(cloud_timestamp, cloud_timestamp))
-
-    @retry(stop=stop_after_attempt(5), wait=wait_exponential(), reraise=True)
-    def _list_objects(
-        self, prefix: str, continuation_token: str | None = None
-    ) -> FilesListResponseDto:
-        params = {"prefix": prefix, "limit": 100, "expand": "url"}
-        if continuation_token is not None:
-            params["continuation_token"] = continuation_token
-        list_url = f"{self.base_url}?{urlencode(params)}"
-        response = requests.get(
-            list_url, headers=self.headers, timeout=REQUEST_TIMEOUT_IN_S
+        os.utime(
+            path=local_file_path,
+            times=(
+                last_modified_on_remote.timestamp(),
+                last_modified_on_remote.timestamp(),
+            ),
         )
-        if response.status_code != 200:
-            response.raise_for_status()
-        return FilesListResponseDto(**response.json())
 
-    def _list_local_files(self, local_dir_path: str) -> dict[str, LocalFileDto]:
+    def _list_local_files(self, local_dir_path: str) -> dict[str, FileDto]:
         """List all files in a local directory."""
         local_files = {}
         for root, _, files in os.walk(local_dir_path):
             for file in files:
                 file_path = os.path.join(root, file)
-                relative_path = os.path.relpath(file_path, start=local_dir_path)
+                relative_path = os.path.relpath(
+                    file_path, start=local_dir_path
+                ).replace(os.sep, "/")
                 file_stats = os.stat(file_path)
-                local_files[relative_path] = LocalFileDto(
+                last_modified = datetime.fromtimestamp(
+                    file_stats.st_mtime, tz=timezone.utc
+                )
+                local_files[relative_path] = FileDto(
+                    name=os.path.basename(file_path),
                     path=file_path,
+                    lastModified=last_modified,
```

Comment:
You can use https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.populate_by_name in the `ConfigDict` to stay with 
```suggestion
                    last_modified=last_modified,
```
as this is currently prone for misusage.